src/Distribution/Nixpkgs/Haskell/Derivation.hs

```diff
@@ -59,6 +59,7 @@ data Derivation = MkDerivation
   , _editedCabalFile            :: String
   , _enableSeparateDataOutput   :: Bool
   , _metaSection                :: Meta
+  , _runHpack                   :: Bool
```

Comment:
The `Derivation` type contains only fields that can be passed as arguments to the generic Haskell builder, but the addition of `runHpack` breaks that rule. Therefore I don't think it's a good idea to add that field in here, especially since it's being mapped into a `preConfigure` hook, which is supposed to be used by *user code*, not by the proper Nixpkgs infrastructure. IMHO, it would be better to create an appropriate entry in `phaseOverrides` by some other means that does not rely on a `runHpack` attribute in `Derivation` itself. I understand it's a *command-line argument* to `cabal2nix`? So maybe that logic should be contained in `main`.

Reply:
I didn't realise that, I will amend the PR tomorrow.